misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -111,14 +109,13 @@ public ModuleCoverage generateReport(JarResolver jarResolver) throws IOException
                     return null;
                 }
             }
-
             execFileLoader.load(executionDataFile.toFile());
-            final IBundleCoverage bundleCoverage = analyzeStructure();
-
+            final CoverageBuilder coverageBuilder = analyzeStructure();
             ModuleCoverage moduleCoverage = new ModuleCoverage();
+            IBundleCoverage bundleCoverage = coverageBuilder.getBundle(title);
             createReport(bundleCoverage, moduleCoverage);
-
-            createXMLReport(bundleCoverage);
+            IBundleCoverage modifiedBundleCoverage = getPartialCoverageModifiedBundle(coverageBuilder);
+            createXMLReport(modifiedBundleCoverage);
```

Comment:
Do we need a local variable here?
```java
createXMLReport(getPartialCoverageModifiedBundle(coverageBuilder));
```

Reply:
https://github.com/ballerina-platform/ballerina-lang/pull/28767/commits/fffea3e92edd4a82053fde14e89e1d52f4c3b4f5